vendor/views/vendor.py

```diff
@@ -364,7 +357,7 @@ class ReceiptDetailView(LoginRequiredMixin, DetailView):
     def get_context_data(self, *args, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        context['payment'] = Payment.objects.get(transaction=self.object.transaction)
+        context['payment'] = self.object.order_item.invoice.payments.filter(success=True).first()
```

Comment:
should these assume there is only one payment, vs returning a list of multiple?

Reply:
The payment information would be the same for all payment success cases. Need to add the roll back to guarantee this.